getnacelle/nacelle-js
# feat: update selected delivery options

## Why are these changes introduced?

Addresses [ENG-7511](https://nacelle.atlassian.net/browse/ENG-7511)

> Implement `cartSelectedDeliveryOptionsUpdate` mutation

Shopify added a `cartSelectedDeliveryOptionsUpdate` mutation in the `2022-10` version of the Shopify Storefront API. By implementing a `cartSelectedDeliveryOptionsUpdate` in `@nacelle/shopify-cart@1.x`, all `cart*` mutations in the latest (`2022-10`) version of the Shopify Storefront API will be available in the `cartClient`.

## What is this pull request doing?

- Adds a new `cartSelectedDeliveryOptionsUpdate` method to the `cartClient`.
- Refactors `packages/shopify-cart/src/client/index.ts` to replace repetitive parameter assignments with `...methodConfig(),`.

## How to test

1. Check out the `ENG-7511-feature-cartSelectedDeliveryOptionsUpdate` branch.
2. If needed, `npm run bootstrap` from the monorepo root.
3. Navigate to the `packages/shopify-cart` directory.
4. `npm run codegen` - type generation should complete successfully and result in no uncommitted changes.
5. `npm run build` - the package should build without errors.
6. `npm run test:ci` - all tests should pass with 100% coverage.

## Local end-to-end testing results

In a local Vite x React project, I found that the `cartSelectedDeliveryOptionsUpdate` payload was sent as expected.

I'm not sure what the process is for creating a [`CartDeliveryGroup`](https://shopify.dev/api/storefront/2022-10/objects/CartDeliveryGroup). I can't test this fully without a valid `deliveryGroupId` + `deliveryOptionHandle` pair, but I am reasonably confident that this should work as expected, given that:
1. The GraphQL payload contains the `selectedDeliveryOptions`.
2. When using `selectedDeliveryOptions: []`, there are no errors.

https://user-images.githubusercontent.com/5732000/197899566-8e79a88f-7700-4c1b-b6c6-2295841359f4.mov

## Checklist

- [x] Created a [changeset](https://github.com/changesets/changesets/blob/main/docs/intro-to-using-changesets.md#using-changesets).
